### ISSUE TITLE
[Bug]Monster out of syncin in Mmo 1.63h

### ISSUE THREAD

**Issue by moepi2k** [suriyun-production/mmorpg-kit-docs#901]
Monster has diffrent position in sight of clients when chasing player

**Comment by moepi2k** [comment]
https://gyazo.com/64900dca53948069c631e70b6744eba9

here a video.
like you can see for the client the monster attack from far away. but at server the monster is next to player. when the player go to the "visual" enemy and try attack you wont hit him because on server monster has diffrent location.

for all clients in mmo the enemy looks same. but always on false position

**Comment by insthync** [comment]
You may try change nav mesh agent -> stopping distance to lower value.

**Comment by moepi2k** [comment]
i think this is a network issue. never had this before. i changed nothing on the enemy or agent. 
the enemy is always on diffrent position at clients then on server

https://gyazo.com/7d5d5d36e8d7638bb89f04faf3e7f17d

**Comment by MosesGomez** [comment]
Have the same error, but havent change anything in the nav mesh before. Also just happens at mmo. On single player mobs attack at the right distance.

**Comment by insthync** [comment]
Server always sync position to clients it clients should not receive difference values, you should try change `NavMeshAgent` ->  `stopping distance` and `NavMeshEntityMovement` -> `moveThreshold` to lower value. You may help me check result in `NavMeshEntityMovement` -> `HandleSyncTransformAtClient` but I very sure that clients will receive correct valu. If it's not move, `moveThreshold` may too high.

**Comment by moepi2k** [comment]
i just copy the orcwarrior and replace the model, i changed no values.

therse also a sync issue for player when server auth is on.
when walking and then stop its okey, but when attack while walk, the player stops at diffrent places

https://gyazo.com/73f2711c660c4979e0c8a4337e28b65d

this is also in mmo.
desync stuff is a game breaker, because client want to attack but never hit because on server entities are on diffrent location

**Comment by moepi2k** [comment]
https://gyazo.com/4bdf3d49e47d0a78238cc2e59325c0dd
https://gyazo.com/668c955de877f0f33db128a1c31f9519

**Comment by insthync** [comment]
So problem occurring when attacking?, is your attacking animation is root motion animation?

**Comment by MosesGomez** [comment]
To me the problem is on 2D. https://www.youtube.com/watch?v=YwCM8iDHVEo

**Comment by moepi2k** [comment]
No its no rootmotion,just in place, in singleplayer it works great, enemy stop infront of player then start attacking

**Comment by moepi2k** [comment]
> So problem occurring when attacking?, is your attacking animation is root motion animation?

Ah you mean for player, no player animation also is in place and not rootmotion

**Comment by moepi2k** [comment]
this is singleplayer
https://gyazo.com/d8218fcf6822b161ce132c418ccc8a26

**Comment by moepi2k** [comment]
btw this is tested with non secure.

as you can see. the player stops where the animations beginns. (in sight from other clients on false location) and after animation the player position will be corrected (what its not doing with server auth on)

https://gyazo.com/1d6f84aa04f7ee7d0c2744edcf3d0163

**Comment by moepi2k** [comment]
Ok figured it out, really weird issue here. The problem wasthe scriptable on the enemy prefab. It was attached but broken somehow. I created a new one with exactly same values and nlw enemy works. Why does it always lost references?

Another thing, today we made e small pvp test with the game, tested non secure movement.

Result: pvp is unplayable, 70: of the hits (in ciew of client that should be hitted) are not registered by server because the enemy was already out of range. At clientside the hit was registered bjt no dmg.

Thus is really a game breaker. In other games they use mechanics like lag compensations to check if the hit was legit when the eneny was there ain the past.

Is a mechanic like that planned in future?
